### PR TITLE
DynamoDB Backup (Point-in-time-recovery)

### LINKED ISSUE #284: Add backup strategy to Dynamo
## Type of Issue:

- [x] Enhancement: New feature or request

#### Issue Creator Checklist

- [x] This issue has been thoroughly documented below; a developer should be able to understand the issue by reading it.

## Background

Currently Dynamo is not configured to run any sort of backups or retention. Given that most government project have this as a requirement this would be a good capability to build in, not just for government projects but good practice in general.

## Proposed Approach

A solution has been pushed to the Seds repo. Review the solution and see what is applicable to be pulled into the quickstart. 
Link to the PR: 
https://github.com/CMSgov/cms-mdct-seds/pull/322
Consider this option:
https://aws.amazon.com/dynamodb/pitr/

Confirm with Eric what a required retention period for a production DB is.

## Value

Out of the box Dynamo backup strategy allowing projects to be data retention compliant 

## AC:

Dynamo will be set up to be backed up and backups retained each at a configurable value
Environments that are backed up will be configurable, dev environments should not be backed up and retained.

### PR DESCRIPTION
**Pull request by joeaero19**
## Purpose
This PR enables backing up dynamodb by turning on point-in-time-recovery on dynamodb configuration. This sets the backup retention period to 35 days (defaul value that cannot be changed).

#### Linked Issues to Close
Closes #284 

## Approach
Upon researching the several ways to backup the dynamodb databse, the most efficient method with minimal configuration that stood out was point-in-time-recovery which is easy to setup and can creates an incremental backup every second and has a retention period of 35 days. The minimum backup retention requirement for CMS is 3 weeks. 

## Learning

https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/PointInTimeRecovery.html

#### Pull Request Creator Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/database/serverless.yml [hunk 1]
```diff
@@ -15,6 +15,7 @@ custom:
       - master
       - val
       - production
+  dynamodbBackupEnabled: ${ssm:/configuration/${self:custom.stage}/dynamodbBackup/enabled~true, ssm:/configuration/default/dynamodbBackup/enabled~true}
```
Comment (mdial89f) [inline services/database/serverless.yml:18]: I'd like this to be as is, but with one additional entry...
```
dynamodbBackupEnabled: ${ssm:/configuration/${self:custom.stage}/dynamodbBackup/enabled~true, ssm:/configuration/default/dynamodbBackup/enabled~true, true}
```
This way, the default setting with no SSM parameters set is true/enabled.
This way, there is no requirement to set this SSM parameter (so change the readme above too).
But, it is fully configurable via SSM if necessary

**Review by mdial89f — CHANGES_REQUESTED** [review]

File: services/database/serverless.yml (hunk 1 above)
Comment (mdial89f) [reply services/database/serverless.yml:18]: Also, do the checklist check boxes if you would, which should guide you to assigning this PR, adding reviewers, and adding it to the proejct board.

Comment (mdial89f) [reply services/database/serverless.yml:18]: @joeaero19 Taking a step back, I'm not sure why we wouldn't enable PITR for all environments.  I'm not sure a configuration override is even called for here.

I vote hardcode it:
```
PointInTimeRecoveryEnabled: true
```

Comment (joeaero19) [reply services/database/serverless.yml]: @mdial89f I set PointInTimeRecoveryEnabled to true by hardcoding it.

**Review by mdial89f — APPROVED** [review]

**Pull request merged**